src/Foundation/NSArray.cs

```diff
@@ -63,7 +63,33 @@ public static NSArray FromNSObjects (int count, params INativeObject [] items)
 		{
 			return FromNativeObjects (items);
 		}
+		public static NSArray FromNSObjects<T> (params T [][] items) where T: class, INativeObject
+		{
+			if (items == null)
+				return null;
+			
+			var ret = new NSMutableArray ((nuint)items.Length);
+			foreach (var t in items)
+				ret.Add (NSArray.FromNSObjects (t));
 
+			return ret;
+		}
+		public static NSArray FromNSObjects<T> (T [,] items) where T: class, INativeObject
+		{
+			if (items == null)
+				return null;
+			
+			var width = items.GetLength (0);
+			var height = items.GetLength (1);
+			var ret = new NSMutableArray ((nuint)(width * height));
```

Comment:
This won't do what the API you're trying to do expects, because it expects an NSArray of NSArray of NSView, not an NSArray of NSView.

I think the easiest is to convert to a `T[][]` and then call that overload.

```csharp
var width = items.GetLength (0);
var height = items.GetLength (1);
var array = new T [height][];
for (var y = 0; y < height; y++) {
    var row = new T [width];
    for (var x = 0; x < width; x++) {
        row [x] = items [x, y];
    }
    items [y] = row;
}
return FromNSObjects (items);
```


```